app/code/Magento/Catalog/Block/Product/ListProduct.php

```diff
@@ -355,6 +355,7 @@ public function getIdentities()
         }
 
         foreach ($this->_getProductCollection() as $item) {
+            // phpcs:ignore Magento2.Performance.ForeachArrayMerge
             $identities = array_merge($identities, $item->getIdentities());
         }
```

Comment:
If possible not to use '// phpcs:ignore Magento2.Performance.ForeachArrayMerge'

>      $arraysItemsToMerge = [];
>      foreach ($this->_getProductCollection() as $item) {
>          $arraysItemsToMerge[] = $item->getIdentities();
>      }
>      $identities = array_merge($identities, ...$arraysItemsToMerge);

Reply:
Hi @engcom-Echo,
Should we leave this part of the code as it was? We did not change it but because of this, there is a failed test. How we should proceed here? Any suggestions?
Thanks!

Reply:
Using 'array_merge' function in a 'for/foreach/while'.
It's a very bad practice because it's a performance killer (especially in memory).

I think if we can change it for the better, it will be very good.

Reply:
@engcom-Echo why should we change it in the scope of this PR if we didn't introduce it or modified it in the scope of this PR?

Reply:
@eduard13

Reply:
Hi @engcom-Echo, I have briefly checked, and looks like we aren't able replace it with anything. I was checking maybe we could use `+` operator here, but it may bring some unexpected behavior.
The main reason, being that `+` will remove the duplicated array values, in comparison with `array_merge` which just pushes the new array at the end of another one.

So I'd suggest to proceed with another investigation (separately), of how and where exactly these identities are used, and see if we are able to remove the duplicated array values.

What do you think?
Thank you.

Reply:
> So I'd suggest to proceed with another investigation (separately)....

Here I agree with you. Thanks for the discussion.